src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -544,12 +544,33 @@ where
             let pkg = Box::new(source).download_now(pkgid, config)?;
             Ok(pkg)
         }
-        None => bail!(
-            "could not find `{}` in {} with version `{}`",
-            dep.package_name(),
-            source.source_id(),
-            dep.version_req(),
-        ),
+        None => {
+            let version: String = dep.version_req().to_string();
+            match PackageId::new(dep.package_name(), &version[1..], source.source_id()) {
```

Comment:
I'm a little uncomfortable just blindly stripping the first character from the version req.  Perhaps only do the check if `version_req().is_exact()`?  I think that should guarantee that it is `=` followed by an exact version.

Which reminds me that this doesn't help for non-specific versions (like `*` or `~2`). That's probably a bigger can of worms, though.